board/opentrons/ot2/rootfs-overlay/etc/systemd/system/dropbear.service.d/create-host-key-directory.conf

```diff
@@ -0,0 +1,8 @@
+# Dropbear stores its host keys in /etc/dropbear, which we symlink to
+# /var/lib/dropbear for persistence across reboots and upgrades.
+
+[Service]
+
+# Note "=-" to tolerate failures, to let Dropbear fall back to ephemeral
+# host keys.
+ExecStartPre=-mkdir -p /var/lib/dropbear
```

Comment:
This unfortunately isn't really a fallback. If dropbear doesn't have an `/etc/dropbear` that is a writable directory, it won't start at all. So if this `mkdir` fails, then ssh won't work anymore - dropbear doesn't generate keys in memory per session, it always writes them to disk somewhere. 

That said if mounts failed then a _lot_ of stuff isn't going to work. The failure case we want to handle is really that a user accidentally created or removed the dir. So maybe just removing this will work, or adding some other handler to remove the dir if the create fails will be ok.

Reply:
> If dropbear doesn't have an /etc/dropbear that is a writable directory, it won't start at all.

Gah. Okay, I see two options, then.

1. Try harder to provide a tmpfs fallback if /var/lib/dropbear can't be made to be writable. This keeps SSH working in a degraded state on crippled systems. This would take some more complex machinery. 
2. Call it a day after trying to create /var/lib/dropbear if it doesn't exist. SSH will work if users delete the directory, but not if there's a more serious filesystem error. (If we go with this, we can implement it with `StateDirectory=dropbear` instead of the `mkdir -p`.)

I'm leaning towards option 2. Can we even upload SSH keys without a working `/var`?

> So maybe just removing this will work, or adding some other handler to remove the dir if the create fails will be ok.

Could you clarify this? Why would we remove the directory?

Reply:
Bad phrasing on the delete thing - if there was a file named /var/lib/dropbear you could remove it and call it a day.

We can definitely make a `StateDirectory` dropin, that's a good way to do it.